pkg/standalone-utils/test/SiloWrapping.test.ts

```diff
@@ -0,0 +1,811 @@
+            await relayer.connect(senderUser).multicall([encodeUnwrap(tokenSender, tokenRecipient, amount)])
+          ).wait();
+          const relayerIsSender = TypesConverter.toAddress(tokenSender) === relayer.address;
+
+          if (!relayerIsSender) {
+            expectTransferEvent(
+              receipt,
+              {
+                from: TypesConverter.toAddress(senderUser),
+                to: TypesConverter.toAddress(relayer),
+                value: amount,
+              },
+              sDAI
+            );
+          }
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayer),
+              to: ZERO_ADDRESS,
+              value: amount,
+            },
+            sDAI
+          );
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayerIsRecipient ? mockSilo : relayer),
+              to: TypesConverter.toAddress(relayerIsRecipient ? relayer : senderUser),
+              value: amount,
+            },
+            DAI
+          );
+        });
+
+        it('stores unwrap output as chained reference', async () => {
+          await relayer
+            .connect(senderUser)
+            .multicall([encodeUnwrap(tokenSender, tokenRecipient, amount, toChainedReference(0))]);
+
+          await expectChainedReferenceContents(relayer, toChainedReference(0), amount);
+        });
+
+        it('unwraps with chained references', async () => {
+          await setChainedReferenceContents(toChainedReference(0), amount);
+
+          const receipt = await (
+            await relayer
+              .connect(senderUser)
+              .multicall([encodeUnwrap(tokenSender, tokenRecipient, toChainedReference(0))])
+          ).wait();
+
+          const relayerIsSender = TypesConverter.toAddress(tokenSender) === relayer.address;
+          if (!relayerIsSender) {
+            expectTransferEvent(
+              receipt,
+              {
+                from: TypesConverter.toAddress(tokenSender),
+                to: TypesConverter.toAddress(relayer),
+                value: amount,
+              },
+              sDAI
+            );
+          }
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayer),
+              to: ZERO_ADDRESS,
+              value: amount,
+            },
+            sDAI
+          );
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayerIsRecipient ? mockSilo : relayer),
+              to: TypesConverter.toAddress(relayerIsRecipient ? relayer : senderUser),
+              value: amount,
+            },
+            DAI
+          );
```

Comment:
This should actually be two steps:
1. **always**: `mockSilo -> relayer`
2. **if relayer is not recipient**: `relayer -> tokenSender` **(NOT senderUser)**